src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -499,32 +499,48 @@ internal static void CreateSymbolicLink(string path, string pathToTarget, bool i
                 }
 
                 Span<byte> bufferSpan = new(buffer);
-                success = MemoryMarshal.TryRead(bufferSpan, out Interop.Kernel32.REPARSE_DATA_BUFFER rdb);
+                success = MemoryMarshal.TryRead(bufferSpan, out Interop.Kernel32.SymbolicLinkReparseBuffer rbSymlink);
                 Debug.Assert(success);
 
-                // Only symbolic links are supported at the moment.
-                if ((rdb.ReparseTag & Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_SYMLINK) == 0)
-                {
-                    return null;
-                }
```

Comment:
- `ReparseTag` is not a flag, so the bitwise operation was unnecessary.
- The if logic is now reversed, so that we only return `null` if the link is not a symlink or a mount point.